src/Router/ParameterResolvingRouter.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace eLife\Journal\Router;
+
+use BadMethodCallException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\Matcher\RequestMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouterInterface;
+
+final class ParameterResolvingRouter implements RouterInterface, RequestMatcherInterface
+{
+    private $router;
+    private $parameterResolver;
+
+    public function __construct(RouterInterface $router, ParameterResolver $parameterResolver)
+    {
+        $this->router = $router;
+        $this->parameterResolver = $parameterResolver;
+    }
+
+    public function setContext(RequestContext $context)
```

Comment:
this is a Decorator, a usual solution to make it shorter is to extract the delegating methods that don't do anything into a RouterDecorator superclass, so that this one only defines the interesting bits